matchmake-extension/utility.go

```diff
@@ -0,0 +1,32 @@
+package matchmake_extension
+
+import (
+	//nex "github.com/PretendoNetwork/nex-go"
+	match_making "github.com/PretendoNetwork/nex-protocols-go/match-making"
+	//"github.com/PretendoNetwork/plogger-go"
+    "reflect"
+	"math"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	"fmt"
+)
+
+func FindSearchMatchmakeSession(searchMatchmakeSession match_making.MatchmakeSession) int {
+	returnSessionIndex := math.MaxUint32
+	//this portion finds any sessions that match the search session. It does not care about anything beyond that, such as if the match is already full. This is handled below.
+	candidateSessionIndexes := []int{}
+	for index, session := range common_globals.Sessions {
+		if reflect.DeepEqual(session.SearchMatchmakeSession, searchMatchmakeSession) {
+			candidateSessionIndexes = append(candidateSessionIndexes, index)
+		}
```

Comment:
Oh God do we not have a better way to do this besides using reflect? Reflect is super slow most of the time. I'm tempted to suggest adding an `Equals` API to `StructureInterface` much like `Copy` and checking for equal structs that way